libs/designsystem/src/lib/components/list/list.component.spec.ts

```diff
@@ -222,4 +222,59 @@ describe('ListComponent', () => {
       });
     });
   });
+
+  describe('List have 1 element', () => {
+    it(`should apply the CSS class 'first' and 'last' on the first and only element in the list`, () => {
+      spectator.setInput({
+        items: [transactions[0]],
+      });
+      const list = spectator.queryAll('ion-item-sliding');
+
+      expect(list.length).toBe(1);
+
+      expect(list[0].classList).toContain('first');
+      expect(list[0].classList).toContain('last');
+    });
+  });
+
+  describe('List have 2 elements', () => {
+    it(`should apply the CSS class 'first' on the first element and apply 'last' on last element`, () => {
+      spectator.setInput({
+        items: [transactions[0], transactions[0]],
+      });
+      const list = spectator.queryAll('ion-item-sliding');
+
+      expect(list.length).toBe(2);
+
+      //First element
+      expect(list[0].classList).toContain('first');
+      expect(list[0].classList).not.toContain('last');
+
+      //Last element
+      expect(list[1].classList).toContain('last');
+      expect(list[1].classList).not.toContain('first');
+    });
+  });
+
+  describe('List have 3 elements', () => {
```

Comment:
You could also divide it up into separate `it` blocks, so the first `it` block would test the first element, the second `it` block the second element and I think you get the idea 😄 It could look something like this:

```
  describe('List have 3 elements', () => {
    beforeEach(() => {
      spectator.setInput({
        items: [transactions[0], transactions[0], transactions[0]],
      });
      
      it('should apply the CSS class 'first' on the first element.', () => {
        const list = spectator.queryAll('ion-item-sliding');

        expect(list[0].classList).toContain('first');
        expect(list[0].classList).not.toContain('last');
      })
    })
  ... and similar approach for testing the middle and last item
  }
```

The idea here is just to let each `it` block be more succinct and test one very specific thing. But this is just a suggestion and it's totally up to you if you want to keep it the way you already did it, which is also perfectly fine :)

Reply:
Thanks I have tought about it and I will keep my implementation

Reply:
Would be great for readability to implement Marks suggestion. 
This is hard to digest: 
'should apply the CSS class 'first' on the first element and apply 'last' on last element, the element(s) between first and last should neither have 'first' or 'last' applied'

Reply:
Okay noted I can see the logic behind that! I will change it